addon/components/-private/base-pop-menu.js

```diff
@@ -227,14 +236,53 @@ export default class BasePopMenuComponent extends Component {
   }
 
   /**
-   * Gets the list of focusable elements in the popper, focuses on the first one.
-   * Makes sure there is actually something to focus on.
+   * Gets the list of focusable elements in the popper, focuses on the specified one.
+   * Also checks for active focusable elements.
    * TODO: Make this a global helper?
+   * @param {string} position - 'first', 'last', 'previous', 'next'
+   * @param {boolean} active - whether it should try to find an active item first
    */
-  _focusOnFirstFocusableElement() {
+  _focusOnFocusableElement(position = 'first', active = true) {
     let focusableElements = this._getFocusableElementsInPopper();
+    // Get starting index from what is already focused in the popper
+    let targetIndex = Array.from(focusableElements).indexOf(document.activeElement);
+
+    // Don't bother doing anything if there are no focusable elements
     if (focusableElements.length > 0) {
-      focusableElements[0].focus();
+
+      // Determine focus target index
+      if (position === 'first') {
+        targetIndex = 0;
+      } else if (position === 'last') {
+        targetIndex = focusableElements.length - 1;
+      } else if (position === 'next') {
+        targetIndex++;
+        if (targetIndex >= focusableElements.length) {
+          targetIndex = 0;
+        }
+      } else if (position === 'prev') {
+        targetIndex--;
+        if (targetIndex < 0) {
+          targetIndex = focusableElements.length - 1;
+        }
+      }
+
+      let focusElement = focusableElements[targetIndex];
+
+      // If "active" is true, we want to focus on the first active element,
+      // if there is one, instead of just the first regular element.
+      if (active) {
+        // get the first active element match if it exists
+        Array.from(focusableElements).some(function(element) {
```

Comment:
This feels like an odd use of `Array.prototype.some` to me, because we never use the boolean output. If we just want to use the first active element, I think `find` would be more suited... something like:
```js
if (active) {
  let firstActiveElement = Array.from(focusableElements).find(element => { 
    return /(selected|active)/.test(element.className) || element.checked;
  });
  focusElement = firstActiveElement || focusElement;
}
```

Also, it might make more sense for this logic block to live the top of the function. If an active element does exist and the flag is true, all the previous logic to find `targetIndex` was not needed.

Reply:
That seems way better, thanks!

Reply:
ended up splitting it up bc I was trying to fit 2 use cases in one function and it was just too clunky